reposerver/src/main/scala/com/advancedtelematic/tuf/reposerver/db/Repository.scala

```diff
@@ -43,11 +43,15 @@ protected [db] class TargetItemRepository()(implicit db: Database, ec: Execution
 
   def persist(targetItem: TargetItem): Future[TargetItem] = db.run(persistAction(targetItem))
 
-  def delete(repoId: RepoId, filename: TargetFilename): Future[Unit] = db.run {
-    targetItems
-      .filter(_.repoId === repoId).filter(_.filename === filename)
-      .delete
+  def deleteWithComment(filenameComments: FilenameCommentRepository, repoId: RepoId, filename: TargetFilename): Future[Unit] = db.run {
```

Comment:
Usually I'd put `deleteWithComment(filenameComments: FilenameCommentRepository)(..other arguments)` just because filenameComments is just a dependency and you could just move around a partially applied method, but mainly to convey the idea that it's more of a dependency than a normal argument.

Maybe a name like `deleteItemAndComments` or something more actionable like this?